Traceback (most recent call last):
  File "/home/phil/mini38/envs/work/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/phil/mini38/envs/work/lib/python3.8/site-packages/nbclient/client.py", line 1082, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/phil/mini38/envs/work/lib/python3.8/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/phil/mini38/envs/work/lib/python3.8/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/home/phil/mini38/envs/work/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/phil/mini38/envs/work/lib/python3.8/site-packages/nbclient/client.py", line 535, in async_execute
    await self.async_execute_cell(
  File "/home/phil/mini38/envs/work/lib/python3.8/site-packages/nbclient/client.py", line 827, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/home/phil/mini38/envs/work/lib/python3.8/site-packages/nbclient/client.py", line 735, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply['content'])
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import pdb

import numpy as np

from a301.radiation import calc_radiance


def multi_layer_radiance(Tsfc, Temps, tau, the_wavel):
    """
    Find the radiance $L_Î»$ reaching a satllite from an N-level atmosphere
    using Stull equation 8.4
    
    Given N levels (counting surface), and N-1 layers with the following layout:
    
    ============= tau[-1]
    
      Temps[-1]
      
    ============= tau[N-2]
      
      .....
      
    ============= tau[2]
    
      Temps[1]
      
    ============= tau[1]
     
      Temps[0]
      
    ============= Tsfc, tau[0]=0
    
    this function calculates the vertical, upward, monochromatic radiance
    at a given wavelength, passing through the top level N of layer N-1.  It includes
    both surface emission (assuming a black surface at temperature Tsfc)
    
    Parameters
    ----------
    
    Tsfc:  float (Kelvin)
       temperature of the black surface
    
    Temps: numpy vector (Kelvin)
       Layer temperatures with Temps[0] being closest to the surface
       Length is N-1, where N is the number of levels
       
    tau: numpy vector (unitless)
       Level values of the absorption optical thicknesses at this wavelength
       taus[0]=0 is the surface, tau[-1]=tau_tot is the total optical thickness
       lenght is N, the number of levels
       
    the_wavel: float (m)
       wavelength for calc_radiance
       
    Returns
    -------
    
    Radiance: float (W/m^2/m/sr) 
       Radiance from the surface and layers passing through the top level
    
    
    """
    ### BEGIN SOLUTION
    Batm = np.array([calc_radiance(the_wavel, the_temp) for the_temp in Temps])
    Bsfc = calc_radiance(the_wavel, Tsfc)
    tau_tot = tau[-1]
    trans = np.exp(-(tau_tot - tau))
    weights = np.diff(trans)
    sfc_flux = Bsfc * np.exp(-tau_tot)
    atm_flux = np.sum(Batm * weights)
    print(f"{type(Batm)}, {type(weights)}")
    the_flux = sfc_flux + atm_flux
    return the_flux
    ### END SOLUTION
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
[0;32m<ipython-input-1-87677aa524bc>[0m in [0;36m<module>[0;34m[0m
[1;32m      3[0m [0;32mimport[0m [0mnumpy[0m [0;32mas[0m [0mnp[0m[0;34m[0m[0;34m[0m[0m
[1;32m      4[0m [0;34m[0m[0m
[0;32m----> 5[0;31m [0;32mfrom[0m [0ma301[0m[0;34m.[0m[0mradiation[0m [0;32mimport[0m [0mcalc_radiance[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      6[0m [0;34m[0m[0m
[1;32m      7[0m [0;34m[0m[0m

[0;31mModuleNotFoundError[0m: No module named 'a301'
ModuleNotFoundError: No module named 'a301'

